Trad/administration (#51)
glpi-project/doc: source/modules/administration/dictionnaries.rst

Dictionaries
=============

.. |import-rule| image:: images/importrule.png
.. |playrule| image:: images/playrule.png

Dictionary's allow to modify data already existing in GLPI or new data in order to group redundant data.

> Dictionaries are based on GLPI :doc:`rules engine </modules/administration/rules/rulesmanagement>` and are available for some types of items (softwares, suppliers, drop-downs). The rules associated with a dictionary will modify values that are either manually inserted or automatically inserted via an inventory tool or via plugins (for instance CSV file injector).

Reviewer: ```suggestion
Dictionaries are based on GLPI :doc:`rules engine </modules/administration/rules/rulesmanagement>` and are available for some types of items (software, suppliers, drop-downs). The rules associated with a dictionary will modify values that are either manually inserted or automatically inserted via an inventory tool or via plugins (for instance CSV file injector).
```

> Dictionaries are based on GLPI :doc:`rules engine </modules/administration/rules/rulesmanagement>` and are available for some types of items (softwares, suppliers, drop-downs). The rules associated with a dictionary will modify values that are either manually inserted or automatically inserted via an inventory tool or via plugins (for instance CSV file injector).

Reviewer: ```suggestion
Dictionaries are based on GLPI :doc:`rules engine </modules/administration/rules/rulesmanagement>` and are available for some types of items (software, suppliers, drop-downs). The rules associated with a dictionary will modify values that are either manually inserted or automatically inserted via an inventory tool or via plugins (for instance CSV file injector).
```

.. note:: export or import use a XML file format

Configure data dictionaries
----------------------------

A dictionary works the following way:

1. data to be added enters the dictionary;
2. rules engine plays all rules applying to this type of data and stops on first matching rule;
3. modified data is returned by the dictionary and inserted into database.

Button |playrule| (under the list of rules of the dictionary) allows to replay the rules on data already existing in database.

.. warning:: if database is big, the parameter *memory\_limit* in PHP configuration file must be carefully adjusted: processing by a dictionary can be heavy.

.. hint::
   * it is highly recommended to play rules on a test database and to backup database before production launch of the dictionary or before inserting new rules
   * a script is available in directory *scripts* of GLPI installation, named *compute\_dictionnary.php*, that allows to launch dictionnary processing in command line mode; this allows to bypass problems of execution limit and provides a significant speedup

Global dictionaries
~~~~~~~~~~~~~~~~~~~

> Softwares

Reviewer: ```suggestion
Software
```

+++++++++

> The softwares dictionary modifies software data (name, version, manufacturer) in order to complete or merge software's. It is used to make more coherent equivalent softwares for which name is not identical (for example Mozilla Firefox 3.0 and Mozilla Firefox 3.6) or to add a manufacturer if it is not present. 

Reviewer: ```suggestion
The software dictionary modifies software data (name, version, manufacturer) in order to complete or merge software's. It is used to make more coherent equivalent software for which name is not identical (for example Mozilla Firefox 3.0 and Mozilla Firefox 3.6) or to add a manufacturer if it is not present. 
```

> This dictionary allows also to redirect the creation of a software or of a set of softwares in a given entity, by choosing action *Entity* and to select the entity in which to create the software. This functionality can also be used with general option *Entity for software* available in entity configuration.

Reviewer: ```suggestion
This dictionary allows also to redirect the creation of a software or of a set of software in a given entity, by choosing action *Entity* and to select the entity in which to create the software. This functionality can also be used with general option *Entity for software* available in entity configuration.
```

> .. note:: for an optimal management of softwares and licenses in a multi-entity environment, it is possible to use the softwares dictionnary simultaneously with softwares visible in sub-entities and thus to use grouping functionality.

Reviewer: ```suggestion
.. note:: for an optimal management of software and licenses in a multi-entity environment, it is possible to use the software dictionary simultaneously with software visible in sub-entities and thus to use grouping functionality.
```

.. warning:: using action *Add regexp result* on a version must be used with maximum care; indeed, this action is only taken into account when importing data coming from an inventory tool and will be ignored when re-applying dictionary on an existing database.

> As an example, following image describes grouping Mozilla softwares. Criteria are accumulated with AND. 

Reviewer: ```suggestion
As an example, following image describes grouping Mozilla software. Criteria are accumulated with AND. 
```

.. figure:: images/critereMozilla.png

>    :alt: Criteria for grouping Mozilla softwares

Reviewer: ```suggestion
   :alt: Criteria for grouping Mozilla software
```

   :align: center

>    Criteria for grouping Mozilla softwares

Reviewer: ```suggestion
   Criteria for grouping Mozilla software
```

> The following image describes associated action for grouping Mozilla softwares.

Reviewer: ```suggestion
The following image describes associated action for grouping Mozilla software.
```

> The following image describes associated action for grouping Mozilla softwares.

Reviewer: ```suggestion
The following image describes associated action for grouping Mozilla software.
```

.. figure:: images/actionMozilla.png

>    :alt: Action for grouping Mozilla softwares

Reviewer: ```suggestion
   :alt: Action for grouping Mozilla software
```

   :align: center

>    Action for grouping Mozilla softwares
> 
> The following image presents result of grouping Mozilla software: softwares are grouped by type (Mozilla Thunderbird, Mozilla Firefox...) and versions are grouped by type.

Reviewer: ```suggestion
   Action for grouping Mozilla software

The following image presents result of grouping Mozilla software: software are grouped by type (Mozilla Thunderbird, Mozilla Firefox...) and versions are grouped by type.
```

.. figure:: images/resultatMozilla.png

>    :alt: Result of grouping Mozilla softwares

Reviewer: ```suggestion
   :alt: Result of grouping Mozilla software
```

   :align: center

>    Result of grouping Mozilla softwares

Reviewer: ```suggestion
   Result of grouping Mozilla software
```
